src/llm_utils.py

```diff
@@ -0,0 +1,176 @@
+        model="gemini-2.0-flash", contents=prompt
+    )   
+
+    print(response.text)
+    
+    # Parse and format the response
+    review_comments = parse_llm_response(response.text)
+    return review_comments
+
+
+
+"""
+This is the final code given by deepseek to parse the response from the LLM model. It includes errors
+"""
+
+def parse_llm_response(response: str) -> List[Dict]:
+    review_comments = []
+    
+    # Split response into valid comment blocks
+    comment_blocks = response.split('---')
+    
+    for block in comment_blocks:
+        block = block.strip()
+        if not block:
+            continue
+            
+        comment_data = {
+            'path': None,
+            'line': None,
+            'body': ''
+        }
+        
+        # Validate required fields
+        has_file = False
+        has_line = False
+        has_comment = False
+        
+        for line in block.split('\n'):
+            line = line.strip()
+            if line.startswith('FILE:'):
+                comment_data['path'] = line.split('FILE:', 1)[-1].strip()
+                has_file = True
+            elif line.startswith('LINE:'):
+                try:
+                    comment_data['line'] = int(line.split('LINE:', 1)[-1].strip())
+                    has_line = True
+                except (ValueError, IndexError):
+                    continue
+            elif line.startswith('COMMENT:'):
+                comment_data['body'] += line.split('COMMENT:', 1)[-1].strip() + '\n'
+                has_comment = True
+            elif line.startswith('SUGGESTION:'):
+                suggestion = line.split('SUGGESTION:', 1)[-1].strip()
+                if suggestion and suggestion != 'N/A':
+                    comment_data['body'] += f"\n```suggestion\n{suggestion}\n```"
+            elif line:
+                comment_data['body'] += line + '\n'
+        
+        # Only add properly formatted comments
+        if has_file and has_line and has_comment:
+            comment_data['body'] = comment_data['body'].strip()
+            review_comments.append(comment_data)
+        else:
+            print(f"Skipping invalid comment block:\n{block}")
+            
+    return review_comments
+
+
+# def parse_llm_response(response: str) -> List[Dict]:
+#     """
+#     Parse the LLM response and format it into review comments
+#     Returns a list of structured comment objects
+#     """
+#     review_comments = []
+    
+#     # Split response into individual comments
+#     comment_blocks = response.split('---')
+    
+#     for block in comment_blocks:
+#         lines = [line.strip() for line in block.split('\n') if line.strip()]
+#         if not lines:
+#             continue
+            
+#         comment_data = {
+#             'body': '',
+#             'path': None,
+#             'position': 1,  # Default to first line if not specified
+#             'suggestion': None
+#         }
```

Comment:
The commented out `parse_llm_response` function has a different structure and logic. It might be useful to keep it for reference, but ensure it's properly commented out and doesn't interfere with the active code. Consider removing the whole block if no longer needed.

```suggestion
Remove the commented out `parse_llm_response` if not needed. Or, add a comment explaining why it is kept for future reference.
```